src/entity/task/vdaf.rs

```diff
@@ -19,35 +20,66 @@ impl Histogram {
         match self {
             Histogram::Categorical(CategoricalBuckets {
                 buckets: Some(buckets),
+                ..
             }) => buckets.len() as u64,
             Histogram::Continuous(ContinuousBuckets {
                 buckets: Some(buckets),
+                ..
             }) => buckets.len() as u64 + 1,
-            Histogram::Opaque(BucketLength { length }) => *length,
+            Histogram::Opaque(BucketLength { length, .. }) => *length,
             _ => 0,
         }
     }
 
+    pub fn chunk_length(&self) -> Option<u64> {
+        match self {
+            Histogram::Categorical(CategoricalBuckets { chunk_length, .. })
+            | Histogram::Continuous(ContinuousBuckets { chunk_length, .. })
+            | Histogram::Opaque(BucketLength { chunk_length, .. }) => *chunk_length,
+        }
+    }
+
     fn representation_for_protocol(
         &self,
         protocol: &Protocol,
     ) -> Result<AggregatorVdaf, ValidationErrors> {
         match (protocol, self) {
             (Protocol::Dap07, histogram) => {
-                Ok(AggregatorVdaf::Prio3Histogram(HistogramType::Opaque {
-                    length: histogram.length(),
-                }))
+                if let Some(chunk_length) = histogram.chunk_length() {
+                    Ok(AggregatorVdaf::Prio3Histogram(HistogramType::Opaque {
+                        length: histogram.length(),
+                        chunk_length: Some(chunk_length),
+                    }))
+                } else {
+                    let mut errors = ValidationErrors::new();
+                    errors.add("chunk_length", ValidationError::new("internal-error"));
```

Comment:
What does this represent? It looks like this is a "chunk_length" "required" error maybe?

Reply:
If chunk_length is empty in a request to create a task with a VDAF that needs it, then `populate_chunk_length()` should fill it in. If a `None` makes it to `representation_for_protocol()`, then that indicates a programming error. I didn't include that fallback logic in `representation_for_protocol()` itself in keeping with the idea that we should only make the chunk_length choice once. (thus avoiding potential issues with already-stored tasks if we change the chunk length selection algorithm in the future)

Reply:
I think we should either panic here or in some other way make it possible to propagate up an app error so we return a 500 and (eventually) are able to monitor that. Sending a 400 status with "internal error" would be confusing as a user of the api — validation errors are client errors and "internal error" should be a 5xx server error

Reply:
Panicking is probably the easiest choice here — it won't take down the whole server, but an engineer should probably should be low-priority alerted as soon as we encounter this code path if it's truly an internal error

Reply:
Alternatively, we could respond with chunk length required and treat it like "if we can't figure it out for you, we require you to supply it"

Reply:
Looking at this makes me wonder if we should be using `ValidationErrors::merge_all(existing_res, "vdaf", vdaf_res)` instead of passing `&mut ValidationErrors` and optionally constructing two nested `ValidationErrors` in this and other leaf functions. I'll make it panic for now, rather than get into that.